Loop/Managers/StatusExtensionDataManager.swift

```diff
@@ -0,0 +1,95 @@
+        guard let glucoseStore = self.dataManager.glucoseStore else {
+            completionHandler(nil)
+            return
+        }
+
+        dataManager.loopManager.getLoopStatus {
+            (predictedGlucose, _, recommendedTempBasal, lastTempBasal, lastLoopCompleted, _, _, error) in
+            
+            let context = StatusExtensionContext()
+        
+            #if IOS_SIMULATOR
+                // If we're in the simulator, there's a higher likelihood that we don't have
+                // a fully configured app. Inject some baseline debug data to let us test the
+                // experience. This data will be overwritten by actual data below, if available.
+                context.batteryPercentage = 0.25
+                context.reservoir = ReservoirContext(startDate: Date(), unitVolume: 42.5, capacity: 200)
+                context.netBasal = NetBasalContext(rate: 2.1, percentage: 0.6, startDate: Date() - TimeInterval(250))
+                context.eventualGlucose = String(
+                    format: NSLocalizedString("Eventually %@", comment: ""),
+                    String(describing: HKQuantity(unit: HKUnit.milligramsPerDeciliterUnit(), doubleValue: 119.0)))
+            #endif
+
+            let dataManager = self.dataManager
+            
+            context.loop = LoopContext(
+                dosingEnabled: dataManager.loopManager.dosingEnabled,
+                lastCompleted: lastLoopCompleted)
+
+            if let glucose = glucoseStore.latestGlucose {
+                context.latestGlucose = GlucoseContext(
+                    latest: glucose,
+                    sensor: dataManager.sensorInfo)
+            }
+            
+            let (netBasalRate, netBasalPercentage, basalStartDate) = dataManager.loopManager.calculateNetBasalRate()
+            if let rate = netBasalRate, let percentage = netBasalPercentage, let startDate = basalStartDate {
+                context.netBasal = NetBasalContext(rate: rate, percentage: percentage, startDate: startDate)
+            }
+            
+            if let reservoir = dataManager.doseStore.lastReservoirValue,
+               let capacity = dataManager.pumpState?.pumpModel?.reservoirCapacity {
+                context.reservoir = ReservoirContext(
+                    startDate: reservoir.startDate,
+                    unitVolume: reservoir.unitVolume,
+                    capacity: capacity)
+            }
+            
+            if let batteryPercentage = dataManager.latestPumpStatusFromMySentry?.batteryRemainingPercent {
+                context.batteryPercentage = Double(batteryPercentage) / 100.0
+            }
+        
+            if let lastPoint = predictedGlucose?.last {
+                context.eventualGlucose = String(
+                    format: NSLocalizedString("Eventually %@", comment: "The subtitle format describing eventual glucose. (1: localized glucose value description)"),
```

Comment:
I'm struggling with this. My original idea was to pass the raw value through and let the extension turn it into a string, but that resulted in a lot of more complex objects (eg. HKQuantity) getting pushed through the context. My thinking was that the status dashboard can deal with a String blob so I'd turn it into a string here. Can't do that in a View/ViewController on the Loop side since I don't want the StatusExtensionDataManager to depend on any of the views. Hence, I'm doing it this way.

Open to ideas for how to proceed. For what it's worth, the current approach is flawed - I haven't figured out how to remove the double's precision from the resulting value. So right now I'm getting context strings like "Eventually 92.1283 mg/dL" on my actual device :-/

I've tried extending HKQuantity to give it a better description, but !@$@# HKQuantity doesn't let you have access to its underlying unit so you can't actually improve the description. Oy. Hoping you have ideas!